packages/@ngtools/webpack/src/refactor.ts

```diff
@@ -48,14 +48,20 @@ export class TypeScriptFileRefactor {
     this._sourceString = new MagicString(this._sourceText);
   }
 
+  /**
+   * Collates the diagnostic messages for the current source file
+   */
   getDiagnostics(): ts.Diagnostic[] {
     if (!this._program) {
       return [];
     }
-
-    return this._program.getSyntacticDiagnostics(this._sourceFile)
-      .concat(this._program.getSemanticDiagnostics(this._sourceFile))
-      .concat(this._program.getDeclarationDiagnostics(this._sourceFile));
+    let diagnostics: ts.Diagnostic[] = this._program.getSyntacticDiagnostics(this._sourceFile)
```

Comment:
Actually this code is being replaced in one of my PR to `this._program.getPreEmitDiagnostics(...)`, which basically does the same thing.